bazel/deps.bzl

```diff
@@ -36,10 +36,10 @@ def stratum_deps():
         http_archive(
             name = "com_github_grpc_grpc",
             urls = [
-                "https://github.com/grpc/grpc/archive/v1.33.2.tar.gz",
+                "https://github.com/grpc/grpc/archive/v1.35.0.tar.gz",
             ],
-            strip_prefix = "grpc-1.33.2",
-            sha256 = "2060769f2d4b0d3535ba594b2ab614d7f68a492f786ab94b4318788d45e3278a",
+            strip_prefix = "grpc-1.35.0",
+            sha256 = "27dd2fc5c9809ddcde8eb6fa1fa278a3486566dfc28335fca13eb8df8bd3b958",
```

Comment:
Not sure if we discussed this already/before, but how about this:
- Merge just the dep updates right _now_ (aka #869 #861 #865). This was a two-way compatible midpoint version right?
- Let it roll out into our deployments, gives us confidence it still works
- Continue iterating on the credentials manager

Reply:
Unfortunately `TlsKeyMaterialsConfig` is removed between 1.33.0 and 1.34.0, where the new APIs appear. So, I don't think there's an easy way just update the dependency. However, I think this PR ready for review.

The review can focus on `stratum/lib/security/credentials_manager.[h,cc]`, which is a pretty small change. The rest of the PR is pretty much just testing.